File: readability/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] >= 'A' && text[i] <= 'z')
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == ' ')
            words++;
    }
    return words;
}

int count_sentences(string text)
{
    int sentence = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
            sentence++;
    }
    return sentence;
}

int main(void)
{
    string text = get_string("Text:");
    double L = count_letters(text) / (double)count_words(text) * 100;
    double S = count_sentences(text) / (double)count_words(text) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: sentimental-readability/readability.py
# TODO
from cs50 import get_string

def grade(s):
    letters, words, sentences = 0, 1, 0
    for letter in s:
        if letter <=  'z' and letter >= 'A':
            letters += 1
        if letter == ' ':
            words += 1
        if letter in ('.', '?', '!'):
            sentences += 1
    L, S = letters * 100 / words, sentences * 100 / words
    index = round(0.0588 * L - 0.296 * S - 15.8)
    return index


text = get_string('Text: ')
index = grade(text)
if index < 1:
    print('Before Grade 1')
elif index >= 16:
    print('Grade 16+')
else:
    print(f'Grade {index}')
